bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/ValueUtils.java

```diff
@@ -59,17 +74,118 @@ public class ValueUtils {
             throw e;
         }
     }
 
+    private static BMap<BString, Object> getPopulatedRecordValue(ValueCreator valueCreator, String recordTypeName) {
+        MapValue<BString, Object> recordValue = valueCreator.createRecordValue(recordTypeName);
+        BRecordType type = (BRecordType) TypeUtils.getReferredType(recordValue.getType());
+        return populateDefaultValues(recordValue, type);
+    }
+
+    public static BMap<BString, Object> populateDefaultValues(BMap<BString, Object> recordValue, BRecordType type) {
+        Map<String, BFunctionPointer<Object, ?>> defaultValues = type.getDefaultValues();
+        if (defaultValues.isEmpty()) {
+            return recordValue;
+        }
+        Strand strand = Scheduler.getStrandNoException();
+        if (strand == null) {
+            try {
+                final CountDownLatch latch = new CountDownLatch(defaultValues.size());
+                populateInitialValuesWithNoStrand(recordValue, type, latch);
+                latch.await();
+            } catch (InterruptedException e) {
+                throw ErrorCreator.createError(
+                        StringUtils.fromString("error occurred when populating default values"), e);
+            }
+        } else {
+            for (Map.Entry<String, BFunctionPointer<Object, ?>> field : defaultValues.entrySet()) {
+                recordValue.populateInitialValue(StringUtils.fromString(field.getKey()),
+                        field.getValue().call(new Object[]{strand}));
+            }
+        }
+        return recordValue;
+    }
+
+    private static void populateInitialValuesWithNoStrand(BMap<BString, Object> recordValue, BRecordType recordType,
+                                                          CountDownLatch latch) {
+        Map<String, BFunctionPointer<Object, ?>> defaultValues = recordType.getDefaultValues();
+        String[] fields = defaultValues.keySet().toArray(new String[0]);
+        invokeFPAsyncIterativelyWithNoStrand(recordValue, defaultValues, fields, "default",
+                Scheduler.getDaemonStrand().getMetadata(), defaultValues.size(), o -> {
+        }, Scheduler.getDaemonStrand().scheduler, latch);
+    }
+
+    public static void invokeFPAsyncIterativelyWithNoStrand(BMap<BString, Object> recordValue,
+                                                            Map<String, BFunctionPointer<Object, ?>> defaultValues,
+                                                            String[] fields, String strandName, StrandMetadata metadata,
+                                                            int noOfIterations, Consumer<Object> futureResultConsumer,
+                                                            Scheduler scheduler, CountDownLatch latch) {
+        if (noOfIterations <= 0) {
+            return;
+        }
+        AtomicInteger callCount = new AtomicInteger(0);
+        scheduleNextFunction(recordValue, defaultValues, fields, strandName, metadata, noOfIterations, callCount,
```

Comment:
Why this function not covered

Reply:
This is because those lines are for record value creation with no strand. The actual use case is covered in std libs like - HTTP, SQL.